src/pages/login-page/index.tsx

```diff
@@ -1,7 +1,54 @@
-import { Link } from 'react-router-dom';
+import { Link, Navigate } from 'react-router-dom';
+import { FormEvent, useState, useMemo, useEffect, ChangeEvent } from 'react';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { loginAction } from '../../store/api-actions';
+import { AuthorizationStatus, CITIES } from '../../const';
+import { clearLoginError } from '../../store/user-slice';
+import './login-page.css';
 import { AppRoute } from '../../app/routes';
 
 function LoginPage(): JSX.Element {
+  const dispatch = useAppDispatch();
+  const authorizationStatus = useAppSelector((state) => state.user.authorizationStatus);
+  const loginError = useAppSelector((state) => state.user.error);
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [clientError, setClientError] = useState<string | null>(null);
+
+  const randomCity = useMemo(() => CITIES[Math.floor(Math.random() * CITIES.length)], []);
+
+  useEffect(() => () => {
+    dispatch(clearLoginError());
+  }, [dispatch]);
+
+  if (authorizationStatus === AuthorizationStatus.Auth) {
+    return <Navigate to={AppRoute.Main} />;
+  }
+
+  const handleSubmit = (evt: FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    const hasLetter = /[a-zA-Z]/.test(password);
```

Comment:
Was it a requirement? If yes, this is alright. Otherwise imagine a situation that user created password not meeting requirements years ago (requirements didn't exist at that times) and now client validation doesn't let him in. Even though password is technically correct